x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1384,4 +1389,32 @@ protected LogicalPlan rule(Aggregate aggregate) {
             return plan;
         }
     }
+
+    /**
+     * Verify that a {@link LogicalPlan} can be executed.
+     *
+     * @param plan The logical plan to be verified
+     * @return a collection of verification failures; empty if and only if the plan is valid
+     *
+     * The following verifications on an optimized logical plan are supported:
+     * 1. EVAL with AUTO_BUCKET expression, its from and to should be foldable
+     */
+    LogicalPlan verify(LogicalPlan plan) throws VerificationException {
```

Comment:
bikeshed:
```suggestion
    LogicalPlan verifyOptimized(LogicalPlan plan) throws VerificationException {
```
Also, we could move this into the Verifier and call it from `EsqlSession.optimizedPlan` to make sure that all verification code (and related stuff) is in one place - but keeping it here is probably fine for now, too.

Reply:
(Moving this to the Verifier is very matter of taste - this location is also fine IMO.)

Reply:
Changed the name, I'll leave it here for now, as the Verifier we have is under analysis, if the optimized plan needs more verifications, we can have it in a dedicated place.